lib/p2p/P2PRepository.ts

```diff
@@ -19,7 +19,7 @@ class P2PRepository {
   }
 
   public addHost = async (host: db.HostFactory): Promise<db.HostInstance> => {
-    return this.models.Host.create(<db.HostAttributes>host);
+    return this.models.Host.upsert(<db.HostAttributes>host , { returning: true })[0];
```

Comment:
This is a bug, it's returning the 0th element of the Promise and not of the result of the promise.

Use this instead:

```javascript
const upsertResult = await this.models.Host.upsert(<db.HostAttributes>host, { returning: true })
return upsertResult[0];
```

Also make sure to remove the extra whitespace between `host` and the comma after it.